src/sequence.rs

```diff
@@ -0,0 +1,154 @@
+//! This module manages project sequences
+
+use std::path::{Path, PathBuf};
+use std::fs;
+
+use git2::Signature;
+use sfml::audio::Music;
+
+use Error;
+use utils;
+
+/// Creates a new user for the project in the current directory.
+/// Assumes the current directory contains a Protonfile.json file.
+///
+/// Impure.
+pub fn new_sequence<P: AsRef<Path>>(name: &str, music_file_path: P) -> Result<(), Error> {
+    
+    // Make sure the name is valid (needed since it will be used in a file path)
+    try!(verify_name(name));
+
+    // Make sure the music file is a valid format
+    try!(verify_file_type(&music_file_path));
+
+    // Make the name of the sequence's directory
+    let mut sequence_dir = String::from("seq_");
+    sequence_dir.push_str(&name);
+    let sequence_dir = sequence_dir;
+
+    // Try to create the sequence's directory
+    // This also throws an error if the directory already exists and is not empty
+    try!(utils::create_empty_directory(Path::new(&sequence_dir))
+        .map_err(|_| Error::DuplicateSequence(name.to_string()) ));
+
+    // Get name of music file from path
+    let music_file_name = try!(utils::file_name_from_path(&music_file_path));
+
+    // Try to copy music file into sequence directory
+    let dest_path = try!(copy_music_file(&music_file_path, &sequence_dir));
+
+    // Get duration of music file
+    let music_duration_sec = try!(get_music_duration_sec(&dest_path));
+
```

Comment:
Can't. sfml needs a &str for `Music::new_from_file` (they are silly and didn't use `AsRef<Path>`). I can move the conversion into the function though.